staging/src/k8s.io/client-go/tools/events/eventseries_test.go

```diff
@@ -206,3 +206,161 @@ func validateEventSerie(messagePrefix string, expectedUpdate bool, actualEvent *
 	}
 
 }
+
+func TestFinishSeries(t *testing.T) {
+	hostname, _ := os.Hostname()
+	testPod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			SelfLink:  "/api/version/pods/foo",
+			Name:      "foo",
+			Namespace: "baz",
+			UID:       "bar",
+		},
+	}
+	regarding, err := ref.GetPartialReference(scheme.Scheme, testPod, ".spec.containers[1]")
+	if err != nil {
+		t.Fatal(err)
+	}
+	related, err := ref.GetPartialReference(scheme.Scheme, testPod, ".spec.containers[0]")
+	if err != nil {
+		t.Fatal(err)
+	}
+	LastObservedTime := metav1.MicroTime{Time: time.Now().Add(-9 * time.Minute)}
+
+	cachedEvent := &v1beta1.Event{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "foo",
+			Namespace: "baz",
+		},
+		EventTime:           metav1.MicroTime{time.Now()},
+		ReportingController: "eventTest",
+		ReportingInstance:   "eventTest-" + hostname,
+		Action:              "started",
+		Reason:              "test",
+		Regarding:           *regarding,
+		Related:             related,
+		Note:                "some verbose message: 1",
+		Series: &v1beta1.EventSeries{
+			Count:            10,
+			LastObservedTime: LastObservedTime,
+		},
+		Type: v1.EventTypeNormal,
+	}
+	anotherCachedEvent := cachedEvent.DeepCopy()
+	anotherCachedEvent.ReportingController = "another-controller"
+	anotherCachedEvent.EventTime = metav1.MicroTime{time.Now().Add(-7 * time.Minute)}
+	anotherCachedEvent.Series = nil
+	createEvent := make(chan *v1beta1.Event)
+	updateEvent := make(chan *v1beta1.Event)
+	patchEvent := make(chan *v1beta1.Event)
+	testEvents := testEventSeriesSink{
+		OnCreate: func(event *v1beta1.Event) (*v1beta1.Event, error) {
+			createEvent <- event
+			return event, nil
+		},
+		OnUpdate: func(event *v1beta1.Event) (*v1beta1.Event, error) {
+			updateEvent <- event
+			return event, nil
+		},
+		OnPatch: func(event *v1beta1.Event, patch []byte) (*v1beta1.Event, error) {
+			// event we receive is already patched, usually the sink uses it only to retrieve the name and namespace, here
+			// we'll use it directly
+			patchEvent <- event
+			return event, nil
+		},
+	}
+	cache := map[eventKey]*v1beta1.Event{
+		getKey(cachedEvent):        cachedEvent,
+		getKey(anotherCachedEvent): anotherCachedEvent,
+	}
+
+	eventBroadcaster := newBroadcaster(&testEvents, 0, cache).(*eventBroadcasterImpl)
+	go eventBroadcaster.finishSeries()
+	actualEvent := <-patchEvent
```

Comment:
We don't want to wait indefinitely in case of problems.
It should be sth like:
```
select {
case actualEvent := <-patchEvent:
  // do stuff
case <-time.After(wait.ForeverTestTimeout):
  // fail test
}
```

Reply:
Ack